### PR TITLE
[RHPAM-2657] KieServer Scale-Up-From-Zero lifecycle hooks logic broken

### PR DESCRIPTION
**Pull request by rhtevan**
Signed-off-by: Evan Zhang <evan.zhang@redhat.com>

Related JIRAs
https://issues.redhat.com/projects/RHPAM/issues/RHPAM-2657
https://issues.redhat.com/projects/RHPAM/issues/RHPAM-2008
https://issues.redhat.com/browse/JBPM-8585

Related PRs
https://github.com/kiegroup/kie-wb-common/pull/3139

Followed-By PRs
https://github.com/jboss-container-images/rhpam-7-openshift-image/pull/414
https://github.com/jboss-container-images/rhdm-7-openshift-image/pull/329
https://github.com/jboss-container-images/jboss-kie-modules/pull/352
https://github.com/kiegroup/kie-cloud-operator/pull/353

### CODE REVIEW

**Comment by elguardian** [comment]
it cannot be a warn message because it won't allow to connect.

**Comment by rhtevan** [comment]
@elguardian For regular 'ControllerBased' KieServer, BC/WB need to have such check/validation, but it should not be needed for OSSS since OpenShift enhancement enabled BC and KieServer are always having consistent view of between KieServerState and ServerTemplate since those two are simply two different views from the same state data persisted at KieServer ConfigMap.

File: kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/KieServerControllerImpl.java [hunk 1]
```diff
@@ -221,7 +222,15 @@ public synchronized KieServerSetup connect(KieServerInfo serverInfo) {
         return serverSetup;
     }
 
+    private boolean isOpenShiftSupported() {
```
Comment (elguardian) [inline kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/KieServerControllerImpl.java]: please change this for a more generice flag naming like disable server setup validation or the sort.
bringing the environmente here is no good

Comment (rhtevan) [reply kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/KieServerControllerImpl.java]: After a quick discussion, @mswiderski and @elguardian are fine with this approach. @cristianonicolai could help having a quick check and get it merged. Thanks!

**Pull request merged**